File: content/csharp/environment/01-the-sdk-and-a-project/solution/Calc.cs
namespace Lesson;

public static class Calc
{
    public static int Add(int a, int b) => a + b;

    public static string Describe(int n) => n switch
    {
        < 0 => "negative",
        0 => "zero",
        _ => "positive",
    };

    public static double Average(int[] values)
    {
        if (values.Length == 0)
        {
            return 0;
        }
        // (double) on the sum, or the division happens in int and truncates.
        return (double)values.Sum() / values.Length;
    }

    public static string Repeat(string text, int count)
    {
        if (count < 0)
        {
            throw new ArgumentOutOfRangeException(nameof(count), "count must not be negative");
        }
        return string.Concat(Enumerable.Repeat(text, count));
    }
}

File: content/c/environment/01-compiling-and-running/solution/calc.c
#include "calc.h"

int c_max(int a, int b) {
    return a > b ? a : b;
}

int clamp(int value, int lo, int hi) {
    if (value < lo) {
        return lo;
    }
    if (value > hi) {
        return hi;
    }
    return value;
}

int is_even(int n) {
    return n % 2 == 0;
}

const char *sign_of(int n) {
    if (n < 0) {
        return "negative";
    }
    if (n == 0) {
        return "zero";
    }
    return "positive";
}
